### PR TITLE
add self image CRUD service

### PR DESCRIPTION
**Pull request by theodore3131**
- defined basic CRUD operations in a separate selfImageExperimentService.js

### CODE REVIEW

**Review by quanganhhoang — CHANGES_REQUESTED** [review]
could you please remove `console.log` lines? They should not be in production code.

**Comment by theodore3131** [comment]
> could you please remove `console.log` lines? They should not be in production code.

Sure, thanks, will update in a sec

**Review by dustinlo — APPROVED** [review]
All returns are consistent and there is no console.log no more.

File: src/firebase/CRUDServices/selfImageExperimentService.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        if (snapshot.empty) {
+          return;
+        }
+
+        snapshot.forEach( (doc) => {
+          // TODO: add logic here
+        });
+      } catch (err) {
+        // TODO: add logic here
+      }
```
Comment (quanganhhoang) [inline src/firebase/CRUDServices/selfImageExperimentService.js:57]: shouldn't this return SelfImageExperiment[] per documentation? Both try and catch blocks are missing return statements

File: src/firebase/CRUDServices/selfImageExperimentService.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+          .doc(experimentId);
+        const doc = await experimentRef.get();
+        if (!doc.exists) {
+          // TODO: add logic here
+        } else {
+          // TODO: add logic here
+        }
+        return doc;
+      } catch (err) {
+        // TODO: add logic here
```
Comment (quanganhhoang) [inline src/firebase/CRUDServices/selfImageExperimentService.js:78]: branch missing here? If there is an error we are not returning anything

**Review by quanganhhoang — CHANGES_REQUESTED** [review]

File: src/firebase/CRUDServices/selfImageExperimentService.js (hunk 2 above)
Comment (theodore3131) [reply src/firebase/CRUDServices/selfImageExperimentService.js:79]: Updated for review.

**Review by quanganhhoang — APPROVED** [review]

**Pull request merged**